Develop

meridian285 opened the pull request:
использованные библиотеки
selenium 3.141.59, webdrivermanager 5.3.0, junit-jupiter-engine 5.9.1, datafaker 1.8.1, allure-junit5

meridian285 committed:
Добавить все необходимые зависимости и плагины

meridian285 committed:
Добавить BaseTest.java и BasePage.java

BaseTest.java - реализована инициализация и настройки веб драйвера
BasePage - Базовый класс для всех классов page

meridian285 committed:
Добавить MainPage.java класс стартовой загрузки сайта

Добавлен локатор кнопки ManagerLoginButton и методнажатия на нее

meridian285 committed:
Добавить ManagePage.java метод страницы Manage Page

Класс абстрактный, т.к. содержит кнопки навигации доступные с других страниц
Добавить локатор Кнопка перехода на страницу добавления клиента addCustomerButton и метод клика для него clickAddCustomerButton().
Добавить локатор кнопки перехода на страницу списка клиентов customersButton, и метод клина для него clickCustomersButton()

meridian285 committed:
Добавить AddCustomerPage.java, класс ввода клиентов

Добавлен локатор поля ввода FirstName firstNameField
Добавлен локатор поля ввода LastName lastNameField
Добавлен локатор поля ввода PostCode postCodeField
Добавлен локатор кнопки Add Customer
Добавлен метод заполнения полей firstNameField, lastNameField, postCodeField и нажатия кнопки Add Customer
Добавлен метод проверки сообщения о создании клиента
Добавлен метод проверки заполнения полей checkingThatFieldsNotFilled

meridian285 committed:
Добавить Customer.java, DataGenerator.java классы для генерации рандомных значений

В класс DataGenerator добавлен метод генерации случайных значений firstName, lastName, postCode
Customer класс с нужными полями для генерации

meridian285 committed:
Добавить CustomersPage.java класс для страницы клиентов

Добавлен локатор поля поиска searchField и метод для поиска findText()
Добавлен локатор для списка клиентов listCustomers
Добавлен метод проверки checkFindCustomers(), аргументом принимает 3 поля String firstName, String lastName, String postCode и по ним проверят найден ли клиент
Добавлен метод проверки сортировки таблицы по первому столбцу checkSortingTable(), метод сверяет два списка до сортировки и после, сортирует в обратном порядке
Добавлен метод получения списков всех клиентов getListCustomers(), метод  переводит список элементов в строковый список и разбивает его на подсписки, один подсписок хранит данные одного клиента. метод privat для использования только внутри класса

meridian285 committed:
Добавить CreateCustomerTests.java тестовый класс для тестов связанных с созданием клиентов

Добавлен тест createCustomerValidCredentials(), позитивный тест на создание клиента
Добавлен тест createCustomerWithoutFirstName(), тест на создание клиента без поля First Name
Добавлен тест createCustomerWithoutLastName(), тест на создание клиента без поля Last Name
Добавлен createCustomerWithoutPostCode(), тест на создание клиента без поля Post Code

meridian285 committed:
Добавить тестовый класс CustomersTests для тестов связанных со страницей Customers

Добавлен тестовый метод checkSearchCustomer() для теста на поиск по First Name
Добавлен тестовый метод checkSortingTable() для сортировки таблицы по имени First Name

meridian285 committed:
Добавить в тестовый класс CreateCustomerTests параметризованные тест createCustomerParamTest()

Тестовый класс принимает значения полей в виде строки и ожидаемый результат в виде логической величины
firstName|lastName|postCode|boolean
Добавить метод AfterEach - public void clearCash()
Добавить метод checkMessageCreateCustomerParam() в класс AddCustomerPage. Метод с аргументом boolean(задается в параметризованных данных) делает проверку: появляется ли сообщение о удачной регистрации

meridian285 committed:
Сделать рефакторинг в методе checkFindCustomers() в классе CustomersPage

Сократился код за  счет переиспользования getListCustomers(). Теперь метод получает сразу разбитые подсписки в строке

meridian285 committed:
Добавить файл test-data.csv, файл для передачи параметров параметризованного теста

meridian285 committed:
Добавлена зависимость org.apache.maven.surefire

meridian285 committed:
Добавить изменены 3 локатора, добавлен метод проверки дубликатов

изменены локаторы firstNameField, lastNameField,postCodeField

meridian285 committed:
Добавить тестовые данные в файл для ввода тестовых данных

meridian285 committed:
Добавить удалил зависимости org.slf4j

meridian285 committed:
Добавить тестовый метод для тестирования создания дубликата клиента

meridian285 committed:
Добавить файл с тест кейсами

sun6r0 commented on src/test/java/com/globalsqa/angularJs_protractor/bankingproject/pages/AddCustomerPage.java:
```diff
@@ -0,0 +1,77 @@
+        String actualAlertMessage= driver.switchTo().alert().getText();
+        Assertions.assertTrue(actualAlertMessage.contains("Please check the details. Customer may be duplicate."),
+                "Ожидалось сообщение об ошибке, нельзя создать дубликат");
+        driver.switchTo().alert().accept();
```
Как правило, ассерты рекомендуют выполнять в тестах, а не в пейджах. Тест становится менее информативным, если ассерты скрыты где-то в другом месте,. Там, где это возможно, должно быть очевидно, что проверяет тест, это лучше всего достигается, когда ассерт находится непосредственно в нем.

Мы используем паттерн page object, чтобы предоставить возможность повторного использования методов на странице, а размещение ассертов внутри вметодов пейджа может ограничить их гибкость в переиспользовании, Например, ты в этих методах захардкодил сообщения об ошибке. А мог сделать один метод в пейдже, который достает нам текст из алерта, а уже в каждом тесте сравнивать его с тем сообщением, которое этот тест должен проверять. Так и кол-во методов в пейджах сократится и в тестах будет прозрачна цель нашей проверки

meridian285 committed:
Добавить файл README.md

meridian285 committed:
Добавить 2 метода clickAlertAccept() и getTextAlertMessage()

clickAlertAccept() кликает ОК
getTextAlertMessage() извлекает текст сообщения

meridian285 committed:
Добавить изменения в метод listSortedBySite(), добавлен метод sortingListInDescending()

Переименован, проверка перенесена в тестовый класс, поэтому теперь метод возвращает список для проверки
Метод sortingListInDescending() - сортирует список в обратном порядке полученный из sortingListInDescending()

meridian285 committed:
Добавить изменения в методы checkingThatFieldsNotFilled(), checkMessageCreateCustomerParam() удалить метод checkMessageCreateDuplicateCustomer()

Метод checkingThatFieldsNotFilled(): убрана проверка из метода, теперь он возвращает проверяет есть ли сообщение и возвращает boolean нужное для проверки в тестовом методе

В методе checkMessageCreateCustomerParam() такие же изменения

meridian285 committed:
Добавить во все тестовые классы проверки, ранее они были в page классах

meridian285 committed:
Добавить во все тестовые классы проверки, ранее они были в page классах

meridian285 committed:
Добавить метод удаления clickDeleteButton()

meridian285 committed:
Добавить изменения в названиях, в параметризованный тест добавлен метод удаления списка клиентов

meridian285 committed:
Добавить изменения в названиях

meridian285 committed:
Добавить файл allure.properties с настройкой allure.results.directory=target/allure-results

немного изменен pom

meridian285 committed:
Добавить pom паралельные тесты 2 потока

немного изменен pom
